src/shared/parameter_v2/scripts/config_yaml_loader.py

```diff
@@ -0,0 +1,288 @@
+                    ) from None
+
+        return raw_config_metadata
+
+    @staticmethod
+    def __validate_config_metadata(config_metadata: dict):
+        """Validates the config_metadata that was loaded against the
+        dynamic_parameter_schemas and then checks for duplicate includes
+        and duplicate parameters in the same config.
+
+        :raises: ConfigYamlMalformed
+        :raises: ConfigSchemaViolation
+        :param config_metadata: Metadata describing params and config includes
+        :type config_metadata: dict
+
+        """
+
+        for config_file, metadata in config_metadata.items():
+
+            if "include" in metadata:
+
+                # check schema
+                try:
+                    jsonschema.validate(metadata["include"], INCLUDE_DEF_SCHEMA)
+                except jsonschema.exceptions.ValidationError as jsval:
+                    raise ConfigYamlSchemaViolation(
+                        "Schema violation in {}: {}".format(config_file, jsval)
+                    ) from None
+
+                # check duplicates
+                if len(metadata["include"]) > len(set(metadata["include"])):
+                    raise ConfigYamlMalformed(
+                        "Duplicate include detected in {}".format(config_file)
+                    )
+
+                # check that included yaml is defined elsewhere
+                for included_yaml in metadata["include"]:
+                    if included_yaml not in config_metadata.keys():
+                        raise ConfigYamlMalformed(
+                            "definition could not be found for {} in {}".format(
+                                included_yaml, config_file
+                            )
+                        )
+
+            if "parameters" in metadata:
+
+                # check schema
+                try:
+                    jsonschema.validate(metadata["parameters"], PARAM_DEF_SCHEMA)
+                except jsonschema.exceptions.ValidationError as jsval:
+                    raise ConfigYamlSchemaViolation(
+                        "Schema violation in {}: {}".format(config_file, jsval)
+                    ) from None
+
+                # Get all parameter names as a list, the parameter type comes
+                # first and the name follows in the dictionary. If the schema
+                # check above succeeded, its safe to assume the "name" key
+                # exists in the parameter dictionary
+                param_names = [
+                    list(param_entry.values())[0]["name"]
+                    for param_entry in metadata["parameters"]
+                ]
+
+                # check duplicates
+                if len(param_names) > len(set(param_names)):
+                    raise ConfigYamlMalformed(
+                        "Duplicate parameter detected in {}".format(config_file)
+                    )
+
+                # This is an ugly artifact of how the yaml is defined and loaded
```

Comment:
I think you might be able to use `oneOf` or `anyOf` with jsonschema to achieve this.

https://json-schema.org/understanding-json-schema/reference/combining.html
https://stackoverflow.com/questions/25014650/json-schema-example-for-oneof-objects

Edit: nevermind we need to validate the keyname, don't think this works

Reply:
yeah I spent a good amount of time on trying to see if we can do as much validation through the schema, but this was the one thing that wouldn't work because of that reason